### ISSUE TITLE
not all online numbers captured

### ISSUE THREAD

**Issue by shivtorov** [DEberhardt/TeamsFunctions#79]
this grubs only the first 500 numbers; should use `-ResultSize` to get them all...

https://github.com/DEberhardt/TeamsFunctions/blob/7c5447455227a25fd1bae1d2e2dfd5a5f36a4869/Public/Functions/UserManagement/TeamsCommonAreaPhone/Get-TeamsCommonAreaPhone.ps1#L123

**Comment by DEberhardt** [comment]
Thank you for raising this. Need to test performance impact.
Only creating the global variable to distinguish phone numbers are Direct Routing numbers (not present) or Calling Plan Numbers (present).

work to do :)

**Comment by DEberhardt** [comment]
for some reason, I get a Commandlet invocation error

```powershell
❯ Get-CsOnlineTelephoneNumber -WarningAction SilentlyContinue -ResultSize Unlimited
Cmdlet invocation error
    + CategoryInfo          : NotSpecified: (:) [Get-CsOnlineTelephoneNumber], CmdletInvocationException
    + FullyQualifiedErrorId : Error processing cmdlet request,Microsoft.Rtc.Management.Hosted.Bvd.GetBvdTelephoneNumberCmdlet
    + PSComputerName        : api.interfaces.records.teams.microsoft.com

 David  C:\  ERROR                                                                                                                                         11:36:17  
❯ Get-CsOnlineTelephoneNumber -WarningAction SilentlyContinue -ResultSize 0
The specified ResultSize value 0 is invalid. Please provide a positive number or 'Unlimited'.
    + CategoryInfo          : NotSpecified: (:) [Get-CsOnlineTelephoneNumber], BvdCmdletException
    + FullyQualifiedErrorId : InvalidResultSize,Microsoft.Rtc.Management.Hosted.Bvd.GetBvdTelephoneNumberCmdlet
    + PSComputerName        : api.interfaces.records.teams.microsoft.com
```
continuing testing...

**Comment by DEberhardt** [comment]
Managed to be able to remove the Global variable now.
Replaced with `Get-CsOnlineTelephoneNumber -TelephoneNumber $PhoneNumber -WarningAction SilentlyContinue` now for all query commands.
During testing this has taken 911ms if Object was found, 723ms if PhoneNumber was not found. This should improve execution times across the board (at least for the first match)

`Get-TeamsTenantVoiceConfig` which is counting the MS numbers have had the `-ResultSize` parameter set to 20000.

Thank you.

Is the invocation error one that should be fed back to the product team?

**Comment by shivtorov** [comment]
> for some reason, I get a Commandlet invocation error
> Get-CsOnlineTelephoneNumber -ResultSize Unlimited

`Get-CsOnlineTelephoneNumber -ResultSize ([UInt32]::MaxValue)`

**Comment by DEberhardt** [comment]
thanks. resolved now - Will be published with v21.07
